core/services/scheduler.go

```diff
@@ -194,8 +194,8 @@ func (ot *OneTime) RunJobAt(initiator models.Initiator, job models.JobSpec) {
 }
 
 func expectedRecurringScheduleJobError(err error) bool {
-	switch err.(type) {
-	case RecurringScheduleJobError:
+	switch errors.Cause(err).(type) {
+	case *RecurringScheduleJobError:
```

Comment:
Everywhere I look, I see us returning the value, not the pointer. The `RecurringScheduleJobError#Error()` that makes it comply with `error` interface is also on a value, and not an interface. Why do we need to match on a pointer here instead of just `RecurringScheduleJobError`? Is there a simple test we can implement to verify it?

```go
// RecurringScheduleJobError contains the field for the error message.
type RecurringScheduleJobError struct {
	msg string
}

// Error returns the error message for the run.
func (err RecurringScheduleJobError) Error() string {
	return err.msg
}

...

	if job.Archived() {
		return nil, RecurringScheduleJobError{
			msg: fmt.Sprintf("Trying to run archived job %s", job.ID),
		}
	}

	now := jm.clock.Now()
	if !job.Started(now) {
		return nil, RecurringScheduleJobError{
			msg: fmt.Sprintf("Job runner: Job %v unstarted: %v before job's start time %v", job.ID, now, job.EndAt),
		}
	}

	if job.Ended(now) {
		return nil, RecurringScheduleJobError{
			msg: fmt.Sprintf("Job runner: Job %v ended: %v past job's end time %v", job.ID, now, job.EndAt),
		}
	}


```

Reply:
I haven't figured out a good way of testing this automatically, without a big redesign, the other thing I could do is make the function public and test it that way if you'd prefer. I have tested manually and this definitely works (and this is definitely failing in production).

Reply:
I just don't fully understand the below comment:

> so fixed the expectedRecurringScheduleJobError function to match properly.

Match what exactly? And how did we know it didn't work?

Reply:
Yeah, let's just make `expectedRecurringScheduleJobError` public and have it tested. The rest of `RecurringScheduleJobError` is already public.

Reply:
Added.